lib/src/main/java/io/ably/lib/util/Crypto.java

```diff
@@ -252,8 +258,13 @@ public String getAlgorithm() {
 		 * @return
 		 */
 		private byte[] getIv() {
-			if(iv == null)
-				return encryptCipher.update(emptyBlock);
+			if(iv == null) {
+				try {
+					return encryptCipher.doFinal(emptyBlock);
+				} catch (IllegalBlockSizeException|BadPaddingException e) {
+					e.printStackTrace();
```

Comment:
This should be a `Log.e(TAG, ...`

Reply:
@paddybyers done https://github.com/ably/ably-java/pull/388/commits/d323a4bc8a6ece011cc42eda606ee29364dfafb1